### ISSUE TITLE
reddit.com | Mobile | Annoyance | App promotion

### ISSUE THREAD

**Issue by in4u** [AdguardTeam/AdguardFilters#24955]
***Description***

Website: `reddit.com`

* **Current behaviour**: 

    Sticky footer at the bottom of page on mobile, promoting app usage.

    Initially, there's a big `reddit.com##.nightmode.amp.SnackBar` asking to load the site in app. After clicking no, on subsequent loads it gets replaced with a small `reddit.com##.XPromoPill` promoting usage of app.

* **Expected behaviour**: 

    No app promotion annoyances. The sticky footer (elements highlighted above) should be hidden.

***System configuration***

Information                            | Value
---                                    | ---
Operating system:                      | Android 7.0
Browser:                               | Firefox for Android 63.0
Filters enabled:                       | Annoyances Filter

**Comment by Alex-302** [comment]
1 - can't reproduce. Please show it on screenshot.
2 - reproduced, added

**Comment by in4u** [comment]
@Alex-302 For 1st case, try opening the site in a new private browsing window or after clearing all cookies and site data. Here's the requested screenshot:

![ss](https://user-images.githubusercontent.com/43062185/48223709-c9df9f80-e3bd-11e8-8607-6c1e7a4dd430.png)

**Comment by Alex-302** [comment]
Thanks, added both.

**Comment by in4u** [comment]
@Alex-302 Sorry, I missed reporting an annoyance related to 1st case. On clicking any post, another confirmation box pops up (refer to screenshot below) highlighting the choice of app or browser. After clicking continue, normal flow resumes and content is revealed. This problem of requiring 2 clicks to reach content also happens on the first load only. So, again try this in a new private browsing window or after clearing all cookies and site data. Can you please add a fix for this as well?

![ss](https://user-images.githubusercontent.com/43062185/48260384-4e273680-e441-11e8-8ae2-ca7d224222da.png)

**Comment by Alex-302** [comment]
added `reddit.com##.XPromoPopup`
Will be needed to press twice, but only the very first time. I think it is OK, if you don't know about this issue:)

**Comment by in4u** [comment]
@Alex-302 Thanks!

@okiehsch, @mapx- I had posted here as per the recommendation at https://github.com/uBlockOrigin/uAssets/issues/3971#issuecomment-434160348. All the primary concerns have been resolved. I just want you guys to check if uBO's extended syntax can help get rid of the one-time need to press twice as noted above.

**Comment by okiehsch** [comment]
`reddit.com##+js(addEventListener-defuser.js, click, [native code])` works on my end.

**Comment by in4u** [comment]
@okiehsch It works but creates a new problem. Whereas earlier after a one-time twice click everything returned to normal, with this filter every clicked page loads after a slight but noticeable delay during which some sort of interstitial is displayed indicating slow page load thereby promoting app usage. I managed to grab a couple of screenshots. In both cases, there were different animated page loaders. 1st one was just sort of logo flashing while 2nd one was logo filling up from bottom along with app promotion.

Unrelated to your filter, I just noticed that the first page load is intentionally delayed during which one of these page loaders is also displayed. Maybe this is why your filter isn't able to work flawlessly?

![s1](https://user-images.githubusercontent.com/43062185/48294420-20c2a300-e4aa-11e8-829b-2624575a93e3.png)

![s2](https://user-images.githubusercontent.com/43062185/48294426-27e9b100-e4aa-11e8-9fd8-ec203100dd4c.png)
